network/fabrics_test.go

```diff
@@ -0,0 +1,875 @@
+		}
+
+		if !strings.Contains(err.Error(), "EOF") {
+			t.Errorf("expected error to contain EOF: found %s", err)
+		}
+	})
+
+	t.Run("bad_decode", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId)), getVlanBadeDecode)
+
+		_, err := do(context.Background(), networkClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+
+		if !strings.Contains(err.Error(), "invalid character") {
+			t.Errorf("expected decode to fail: found %s", err)
+		}
+	})
+
+	t.Run("error", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountUrl, "fabrics", "default", "vlans"), getVlanError)
+
+		resp, err := do(context.Background(), networkClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+		if resp != nil {
+			t.Error("expected resp to be nil")
+		}
+
+		if !strings.Contains(err.Error(), "unable to get vlan") {
+			t.Errorf("expected error to equal testError: found %v", err)
+		}
+	})
+}
+
+func TestGetFabric(t *testing.T) {
+	networkClient := MockNetworkClient()
+
+	do := func(ctx context.Context, nc *network.NetworkClient) (*network.Network, error) {
+		defer testutils.DeactivateClient()
+
+		network, err := nc.Fabrics().Get(ctx, &network.GetFabricInput{
+			NetworkID:    fakeNetworkId,
+			FabricVLANID: fakeFabricVlanId,
+		})
+		if err != nil {
+			return nil, err
+		}
+		return network, nil
+	}
+
+	t.Run("successful", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId), "networks", fakeNetworkId), getFabricSuccess)
+
+		resp, err := do(context.Background(), networkClient)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if resp == nil {
+			t.Fatalf("Expected an output but got nil")
+		}
+	})
+
+	t.Run("eof", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId), "networks", fakeNetworkId), getFabricEmpty)
+
+		_, err := do(context.Background(), networkClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+
+		if !strings.Contains(err.Error(), "EOF") {
+			t.Errorf("expected error to contain EOF: found %s", err)
+		}
+	})
+
+	t.Run("bad_decode", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId), "networks", fakeNetworkId), getFabricBadeDecode)
+
+		_, err := do(context.Background(), networkClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+
+		if !strings.Contains(err.Error(), "invalid character") {
+			t.Errorf("expected decode to fail: found %s", err)
+		}
+	})
+
+	t.Run("error", func(t *testing.T) {
+		testutils.RegisterResponder("GET", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId), "networks"), getFabricError)
+
+		resp, err := do(context.Background(), networkClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+		if resp != nil {
```

Comment:
Add a newline above this `resp != nil` since it was part of the test and not standard error handling idioms.